### PR TITLE
Add RegexDatabase in ITs

### PR DESCRIPTION
**Pull request by alban-auzeill**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/regex/ReluctantQuantifierCheck.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ public void visitSequence(SequenceTree tree) {
           .ifPresent(negatedClass -> {
             String newQuantifier = makePossessive(repetition.getQuantifier());
             String message = String.format("Replace this use of a reluctant quantifier with \"%s%s\".", negatedClass, newQuantifier);
-            reportIssue(repetition, String.format(message, negatedClass), null, Collections.emptyList());
+            reportIssue(repetition, message, null, Collections.emptyList());
```
Comment (alban-auzeill) [inline java-checks/src/main/java/org/sonar/java/checks/regex/ReluctantQuantifierCheck.java:67]: Fix this bug making the rule throwing exceptions on some regex containing `^`

File: java-frontend/src/main/java/org/sonar/java/regex/ast/RegexSource.java [hunk 2]
```diff
@@ -61,7 +61,7 @@ public int length() {
     return sourceText.length();
   }
 
-  List<LiteralTree> getStringLiterals() {
+  public List<LiteralTree> getStringLiterals() {
```
Comment (alban-auzeill) [inline java-frontend/src/main/java/org/sonar/java/regex/ast/RegexSource.java:64]: Required by the export script that collects string literals

File: its/ruling/src/test/resources/regex-examples/java-S5994.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+'org.regex-examples:regex-examples:src/main/java/org/regex/examples/RegexDatabase5.java':[
+135,
```
Comment (alban-auzeill) [inline its/ruling/src/test/resources/regex-examples/java-S5994.json:3]: Seems to be a false-positive, I will investigate

Comment (sebastian-hungerecker-sonarsource) [reply its/ruling/src/test/resources/regex-examples/java-S5994.json:3]: The possessive quantifier check should be changed to not apply to repetitions with a fixed upper bound.

**Review by sebastian-hungerecker-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**